packages/wallet/ui/src/components/Offer.jsx

```diff
@@ -103,27 +103,35 @@ const OfferWithoutContext = ({
   const wants = Object.entries(want).sort(([kwa], [kwb]) => cmp(kwa, kwb));
 
   const OfferEntry = (type, [role, { amount, pursePetname }]) => {
-    const value =
-      amount.displayInfo.assetKind === 'nat' ? Nat(amount.value) : amount.value;
+    let valueString = 'unknown';
+    try {
+      const value =
+        amount.displayInfo.assetKind === 'nat'
+          ? Nat(amount.value)
+          : amount.value;
+      valueString = stringifyPurseValue({
+        value,
+        displayInfo: amount.displayInfo,
+      });
+    } catch {
+      console.warn("Couldn't parse amount value for", pursePetname, amount);
+    }
+
```

Comment:
this block is for transforming an amount to a string. I think that would be useful outside this module. It also should be thoroughly tested with different amount cases, including failure. 

```suggestion
    const valueString = stringifyAmount(amount);
```